include/arch/x86/segmentation.h

```diff
@@ -110,18 +110,6 @@ struct __packed task_state_segment {
 	u16_t iomap;
 };
 
-/* Section 3.4.2 of IA architecture SW developer manual, Vol 3. */
-struct __packed segment_selector {
```

Comment:
Why did we remove this?

Reply:
it's not being used, then it violates rule 5.7 - tag names should be a unique identifier. Note that there is an undefined behaviour for this according with c99.

Also it violates the advisory rule 2.3 - project should not have unused type declarations.

Reply:
not the struct, segment_selector is being used as an uint16_t always.
The struct was (my guess) supposed to make easier access its bits directly but this is not happening, otherwise removing it would break the build.

Reply:
yeah, i was just grepping and not paying attention to the context :)